.drone.yml

```diff
@@ -3,31 +3,91 @@ kind: pipeline
 name: pipeline1
 
 steps:
+- name: restore-cache
+  image: plugins/s3-cache
+  settings:
+    pull: true
+    access_key:
+      from_secret: drone_aws_access_key_id
+    secret_key:
+      from_secret: drone_aws_secret_access_key
+    restore: true
+    root: TODO-BUCKET-NAME
+    filename: pipeline1.tar
+
 - name: checkstyle-and-sevntu
   image: maven:3.6.3-adoptopenjdk-8
+  environment:
+    MAVEN_OPTS: "-Dmaven.repo.local=.m2"
   commands:
   - ./.ci/travis/travis.sh checkstyle-and-sevntu
 
 - name: spotbugs-and-pmd
   image: maven:3.6.3-adoptopenjdk-8
+  environment:
+    MAVEN_OPTS: "-Dmaven.repo.local=.m2"
   commands:
   - ./.ci/travis/travis.sh spotbugs-and-pmd
 
+- name: rebuild-cache
+  image: plugins/s3-cache
+  settings:
+    pull: true
+    access_key:
+      from_secret: drone_aws_access_key_id
+    secret_key:
+      from_secret: drone_aws_secret_access_key
+    rebuild: true
+    root: TODO-BUCKET-NAME
+    filename: pipeline1.tar
+    mount:
+    - .m2
+
 ---
 kind: pipeline
 name: pipeline2
 
 steps:
+- name: restore-cache
+  image: plugins/s3-cache
+  settings:
+    pull: true
+    access_key:
+      from_secret: drone_aws_access_key_id
+    secret_key:
+      from_secret: drone_aws_secret_access_key
+    restore: true
+    root: TODO-BUCKET-NAME
+    filename: pipeline2.tar
```

Comment:
why cant we use same cache for all jobs?

Reply:
something like 
https://github.com/checkstyle/checkstyle/blob/9a77689fe9a81a4c1ad0c8f4a923b2b40f1fb360/.github/workflows/pitest.yml#L17 

if drone supports bash variables in yml.

Reply:
> why cant we use same cache for all jobs?

Every pipeline will likely to have different dependencies. Maven downloads only jars required for requested goals. E.i. `compile` will download jar to compile java files and process resources. `test` will also download dependencies with scope `test` (junit, mockito...) an surfire-plugin, `site` don't need `test` but has a lot of own dependencies to generate docs.

Reply:
Having single zip with all dependencies is good, but not critical.

We can start with any working approach and improve in further PRs if required or get to know how.

Reply:
There can be zip file with all "core" dependencies, some job-specific stuff can be downloaded. It will make it easier to keep track of cached entries